site/public/js/calendar.js

```diff
@@ -806,3 +806,23 @@ function colorLegend() {
     });
 }
 
+// Modifies cookies so the correct filtering of courses on the calendar is chosen.
+// param string course_val Id of the course
+// param string display_name display name of the course
+function filter_course(course_val, display_name) {
+    const [courses_val, display_name] = args.split(',');
```

Comment:
What's going on here?  Why aren't you just using the arguments directly instead of trying to dynamically compute them?